jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/LifeCycleBaseTest.java

```diff
@@ -2388,7 +2391,151 @@ public void testCompleteByActiveTasks() {
+            public void afterTaskStartedEvent(TaskEvent event) {
+                assertNull(event.getTask().getTaskData().getTaskInputVariables());
+                assertNull(event.getTask().getTaskData().getTaskOutputVariables());
+                
+                event.getTaskContext().loadTaskVariables(event.getTask());
+                
+                assertNotNull(event.getTask().getTaskData().getTaskInputVariables());
+                assertEquals(1, event.getTask().getTaskData().getTaskInputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskInputVariables().containsKey("input"));
+                
+                assertNull(event.getTask().getTaskData().getTaskOutputVariables());
+            }
+
+            @Override
+            public void afterTaskCompletedEvent(TaskEvent event) {
+                assertNull(event.getTask().getTaskData().getTaskInputVariables());                
+                assertNotNull(event.getTask().getTaskData().getTaskOutputVariables());
+                assertEquals(1, event.getTask().getTaskData().getTaskOutputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskOutputVariables().containsKey("content"));
+                
+                event.getTaskContext().loadTaskVariables(event.getTask());
+                
+                assertNotNull(event.getTask().getTaskData().getTaskInputVariables());
+                assertEquals(1, event.getTask().getTaskData().getTaskInputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskInputVariables().containsKey("input"));
+                
+                assertNotNull(event.getTask().getTaskData().getTaskOutputVariables());                
+                assertEquals(1, event.getTask().getTaskData().getTaskOutputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskOutputVariables().containsKey("content"));
+            }
+
+            @Override
+            public void afterTaskAddedEvent(TaskEvent event) {
+                assertNotNull(event.getTask().getTaskData().getTaskInputVariables());
+                assertEquals(1, event.getTask().getTaskData().getTaskInputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskInputVariables().containsKey("input"));
+                
+                assertNull(event.getTask().getTaskData().getTaskOutputVariables());
+                
+                event.getTaskContext().loadTaskVariables(event.getTask());
+                
+                assertNotNull(event.getTask().getTaskData().getTaskInputVariables());
+                assertEquals(1, event.getTask().getTaskData().getTaskInputVariables().size());
+                assertTrue(event.getTask().getTaskData().getTaskInputVariables().containsKey("input"));
+                
+                assertNull(event.getTask().getTaskData().getTaskOutputVariables());
+            }
```

Comment:
yes, I thought about it a bit and couldn't find much better way which would not make it much less readable thus I stayed with duplicated as we can easily read it 

if you have any suggestion how to improve without loosing too much of readability then I am all in